File: Laboratorio 1/Matrices/matrices.java
import java.util.*;
 public class matrices{
     public static void main (String[]Args){
    	 Scanner scan = new Scanner (System.in);
    	 
    	 System.out.println("ingrese el tamano de la matriz");
    	 int n=scan.nextInt();
    	 
    	 int matriza[][] = new int [n][n];
         int matrizb[][] = new int [n][n];
         int matrizc[][] = new int [n][n];
         int i, j;
         double tiempo_inicial;
         double tiempo_final; 
         double tiempo_total;
         
         
         
         System.out.println("Datos de la Matriz A :");
         
         for (i=0; i<n; i++){
             for (j=0; j<n; j++){
                 System.out.print("Escribir valor " + i + " , " + j + " : ");
                 matriza [i][j]= scan.nextInt();
                }
            }
            
            System.out.println("Datos de la Matriz B :");
            
            for (i=0; i<n; i++){
             for (j=0; j<n; j++){
                 System.out.print("Escribir valor " + i + " , " + j + " : ");
                 matrizb [i][j]= scan.nextInt();
                }
            }
            tiempo_total = 0.0;
            tiempo_inicial = System.currentTimeMillis();
            for (i=0; i<n; i++){
             for (j=0; j<n; j++){
                 matrizc [i][j]= matriza[i][j]*matrizb[i][j];
                }
            }
            tiempo_final = System.currentTimeMillis();
            tiempo_total = tiempo_final - tiempo_inicial;
            
            System.out.println("Matriz resultante de la multiplicación :");
            
    for (i=0;i<n;i++){
            for (j=0;j<n;j++) {
                System.out.print(matrizc[i][j] + " ");
            }
            System.out.println("");
        }
        System.out.println("Tiempo de ejecucion: "+ tiempo_total/1000);

    }
}

File: Laboratorio 1/Matrices/matrices.cpp
#include <iostream>
#include <time.h>
using namespace std;

int main(){
    clock_t t;
    int n;
    cout<<"Ingrese el tamano de la matriz n: "<<endl;
    cin>>n;
    int i, j, k, a[n][n], b[n][n], c[n][n];
    for(i = 0 ; i < n ; i++){
        for(j = 0 ; j < n ; j++){
            cout << "Ingresar Matriz A [" << i << "]["
                << j << "]: ";
            cin >> a[i][j];
        }
    }
    cout<<endl;
    for(i = 0 ; i < n ; i++){
        for(j = 0 ; j < n ; j++){
            cout << "Ingresar Matriz B [" << i << "]["
                << j << "]: ";
            cin >> b[i][j];
        }
    }
  t = clock(); 
  for(i=0;i<n;i++){ 
      for(j=0;j<n;j++){
          c[i][j]=0;
          for(k=0;k<n;k++){
              c[i][j]=c[i][j]+(a[i][k]*b[k][j]);
          }
      }
  }
  t = clock() - t;
  float ttotal = t;
  cout << endl << "Matriz C (Matriz A*B)." << endl;
  cout << endl;
  for(i=0;i<n;i++){ 
      for(j=0;j<n;j++){
          cout << c[i][j] << " ";
          if(j==2)
          cout << endl << endl;
      }
  }
  
  cout<< ("Tiempo de ejecucion: " ,((float)t)/CLOCKS_PER_SEC);
  return 0;
}
